pymeasure/instruments/hp/hp3478A.py

```diff
@@ -361,17 +394,20 @@ def __init__(self, resourceName, **kwargs):
     range = Instrument.control(
```

Comment:
You could also consider using a slightly different approach: you can also write a property where you query the present mode to the instrument and use that to translate the provided range (can be a float in that case as the user doesn't need to pass the mode to the property) to the correct command; see for example the sensitivity property in the DPS7265: https://github.com/pymeasure/pymeasure/blob/78c5ed3bed5ed6ffcbabadeeb13bbc6feb6f5ee7/pymeasure/instruments/signalrecovery/dsp7265.py#L171-L197

Maybe you can also achieve this in a get and set method rather than by writing a complete own property

Reply:
I changed the range property to use instrument.measurment and added a setter-function.